test/get.sh

```diff
@@ -135,10 +135,25 @@ it_can_get_and_set_git_config() {
   test "$(git config --global credential.helper)" == '!true long command with variables $@'
 }
 
+it_returns_tag_when_using_tag_filter() {
+  local repo=$(init_repo)
+  local ref1=$(make_commit $repo)
+  local ref2=$(make_annotated_tag $repo "1.0-staging" "a tag")
+  local ref3=$(make_commit $repo)
+  local ref4=$(make_annotated_tag $repo "1.0-production" "another tag")
+  local ref5=$(make_commit $repo)
+  get_uri_tag_with_tag_filter $repo $ref2 "*-staging" $TMPDIR/destination | jq -e "
```

Comment:
As commented, the `tag_filter` logic has been implemented in #31 

In this test we check that: 
 * When ref is a tag (`1.0-staging` in that test)
 * And `tag_filter` is defined
 * Then the same tag `1.0-staging` is returned.

I refer again to the comment https://github.com/concourse/git-resource/pull/46#issuecomment-208901148, as maybe the simplest solution is always return the input ref passed to in, as the get operation should always checkout the same version.